examples/sktime/flavor.py

```diff
@@ -0,0 +1,639 @@
+mlflow.pyfunc
+    Produced for use by generic pyfunc-based deployment tools and batch inference.
+
+    The `pyfunc` flavor of the model supports sktime predict methods `predict`,
+    `predict_interval`, `predict_proba`, `predict_quantiles`, `predict_var`.
+
+    The interface for utilizing a sktime model loaded as a `pyfunc` type for
+    generating forecasts requires passing an exogenous regressor as Pandas
+    DataFrame to the `pyfunc.predict()` method (an empty DataFrame must be
+    passed if no exogenous regressor is used). The configuration of predict
+    methods and parameter values passed to the predict methods is defined by
+    a dictionary to be saved as an attribute of the fitted sktime model
+    instance. If no prediction configuration is defined `pyfunc.predict()`
+    will return output from sktime `predict` method. Note that for `pyfunc`
+    flavor the forecasting horizon `fh` must be passed to the fit method.
+
+    Predict methods and parameter values for `pyfunc` flavor can be defined
+    in two ways: `Dict[str, dict]` if parameter values are passed to
+    `pyfunc.predict()`, for example
+    `{"predict_method": {"predict": {}, "predict_interval": {"coverage": [0.1, 0.9]}}`.
+    `Dict[str, list]`, with default parameters in predict method, for example
+    `{"predict_method": ["predict", "predict_interval"}` (Note: when including
+    `predict_proba` method the former appraoch must be followed as `quantiles`
+    parameter has to be provided by the user). If no prediction config is defined
+    `pyfunc.predict()` will return output from sktime `predict()` method.
+"""
+import logging
+import os
+import pickle
+import flavor
+
+import pandas as pd
+import yaml
+
+import sktime
+from sktime.utils.multiindex import flatten_multiindex
+
+from mlflow import pyfunc
+from mlflow.utils.requirements_utils import _get_pinned_requirement
+from mlflow.utils.environment import _mlflow_conda_env
+from mlflow.exceptions import MlflowException
+from mlflow.models import Model
+from mlflow.models.model import MLMODEL_FILE_NAME
+from mlflow.models.utils import _save_example
+from mlflow.protos.databricks_pb2 import INVALID_PARAMETER_VALUE
+from mlflow.utils.environment import (
+    _CONDA_ENV_FILE_NAME,
+    _CONSTRAINTS_FILE_NAME,
+    _PYTHON_ENV_FILE_NAME,
+    _REQUIREMENTS_FILE_NAME,
+    _process_conda_env,
+    _process_pip_requirements,
+    _PythonEnv,
+    _validate_env_arguments,
+)
+from mlflow.utils.file_utils import write_to
+from mlflow.utils.model_utils import (
+    _validate_and_copy_code_paths,
+    _validate_and_prepare_target_save_path,
+    _add_code_from_conf_to_system_path,
+    _get_flavor_configuration,
+)
+from mlflow.tracking._model_registry import DEFAULT_AWAIT_MAX_SLEEP_SECONDS
+from mlflow.tracking.artifact_utils import _download_artifact_from_uri
+from mlflow.exceptions import MlflowException
+from mlflow.protos.databricks_pb2 import INTERNAL_ERROR
+
+FLAVOR_NAME = "flavor"
```

Comment:
This shouldn't be called "flavor". "sktime" perhaps.

Reply:
fixed